src/DotVVM.Framework/Configuration/DefaultSerializerSettingsProvider.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using DotVVM.Framework.ViewModel.Serialization;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+
+namespace DotVVM.Framework.Configuration
+{
+    public sealed class DefaultSerializerSettingsProvider
+    {
+        public JsonSerializerSettings Settings { get; private set; }
```

Comment:
Are settings public and mutable on purpose?

Reply:
I am changing it to a `readonly` field

Public modifier is currently not that necessary (it is accessed only from DotVVM.Framework and DotVVM.Framework.Tests). But in future it should be also accessed by generated REST API bindings.

Mutability is not desirable. However, it is hard to ensure immutability. This is mostly due to the fact that the class is designed to have mutable instances. I ran a simple test and measured the size of `JsonSerializerSettings` instance to be ~310 bytes

Reply:
We could either make the field internal (and not modify it ourselves) or create new instance every time (preferably not by deserializing another instance, rather by calling the constructor and setting the properties)

Reply:
> I am changing it to a readonly field

Nah, whatever, I was talking about the `JsonSerializerSettings` instance, not about the private setter - that it ok

Reply:
> I was talking about the JsonSerializerSettings instance, not about the private setter - that it ok

I know. I just realized that a readonly field is sufficient in this case.

> We could either make the field internal (and not modify it ourselves)

I can make it internal for now (at least for the purposes of this specific bugfix). Not modifying it within the framework would be the best approach in my opinion.

> ...  or create new instance every time (preferably not by deserializing another instance, rather by calling the constructor and setting the properties)

The implementation is already changed to create a new instance by calling the constructor when requested.  However, we probably do not want to create a new instance every time someone accesses settings. While I realize that it would resolve this issue quite elegantly, there are problems with this approach:

-  instances are not trivial (~310 bytes)
-  they will be used once and immediately thrown away
-  many instances might be created for each request

While I have not really performed any perf-tests, I feel like it could negatively impact performance.